File: Sources/Client/Private/Sources/Wave/WaveRoom.ts
/** Represents the current state of a wave room */
export const enum WaveRoomState {
    // State waiting, mean waiting for players
    WAITING,
    // State started, mean playing game currently
    PLAYING,
    // State ended, mean already game over
    ENDED,
}

/** Determines if a wave room is public or private */
export const enum WaveRoomVisibleState {
    PUBLIC,
    PRIVATE,
}

/** Indicates if a player is ready to start the wave */
export const enum WaveRoomPlayerReadyState {
    PREPARING,
    READY,
}

File: Sources/Client/private/src/Florr/OwnNative/Wave/WaveRoom.zig
/// Represents the state of the wave room.
pub const State = enum(u8) {
    /// Waiting, meaning waiting for players (not starting).
    waiting,
    /// Playing, meaning playing currently.
    playing,
    /// Ended, meaning game is already over.
    ended,
};

/// Represents the visible state of the wave room.
pub const VisibleState = enum(u8) {
    public,
    private,
};

/// Represents the player ready state in the wave room.
pub const PlayerReadyState = enum(u8) {
    preparing,
    ready,
};
